cli/src/pcluster/models/imagebuilder.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This module contains all the classes representing the Resources objects.
+# These objects are obtained from the configuration file through a conversion based on the Schema classes.
+#
+
+from typing import List
+
+from pcluster import utils
+from pcluster.config.extended_builtin_class import MarkedBool, MarkedTag
+from pcluster.models.cluster import Resource, Tag
+from pcluster.validators.ec2 import BaseAMIValidator
+
+# ---------------------- Image ---------------------- #
+
+
+class Volume(Resource):
+    """Represent the volume configuration for the ImageBuilder."""
+
+    def __init__(self, size: int = None, encrypted: bool = None, kms_key_id: str = None):
+        super().__init__()
+        self.size = size
+        self.encrypted = encrypted
+        self.kms_key_id = kms_key_id
+        self._default_value()
+        # TODO: add validator
+
+    def _default_value(self):
+        if not self.encrypted:
+            self.encrypted = MarkedBool(False)
+
+
+class Component(Resource):
+    """Represent the components configuration for the ImageBuilder."""
+
+    def __init__(self, type: str = None, value: str = None):
+        super().__init__()
+        self.type = type
+        self.value = value
+        # TODO: add validator
+
+
+class Image(Resource):
+    """Represent the image configuration for the ImageBuilder."""
+
+    def __init__(
+        self,
+        name: str,
+        description: str = None,
+        tags: List[Tag] = None,
+        parameter_store_name_output: str = None,
+        root_volume: Volume = None,
+        components: List[Component] = None,
+    ):
+        super().__init__()
+        self.name = name
+        self.description = description
+        self.tags = tags
+        self.parameter_store_name_output = parameter_store_name_output
+        self.root_volume = root_volume
+        self.components = components
+        self._default_value()
+        # TODO: add validator
+
+    def _default_value(self):
+        if not self.tags:
+            self.tags = []
+        self.tags.append(MarkedTag("PclusterVersion", "".join(utils.get_installed_version())))
```

Comment:
Another option is:
`default_tag = Tag("PclusterVersion", "".join(utils.get_installed_version())`
`default_tag.implied = True`
`self.tags.append(default_tag)`
I couldn't do it with built-in classes because Python doesn't allow adding more attribute to build-in classes. But here it is reasonable to utilize the flexibility of Python to just add the `implied` attribute here.
`

Reply:
good idea, done